cirrus/server/tests/test_experiment_recipes.py

```diff
@@ -1,13 +1,87 @@
+import unittest
+from unittest.mock import MagicMock, patch
+
+import requests
+
 from ..cirrus.experiment_recipes import RemoteSettings
 
 
-def test_get_recipes_is_empty():
-    rs = RemoteSettings()
-    assert rs.get_recipes() == []
+class TestFetchRecipes(unittest.TestCase):
+    def get_remote_settings(self):
+        return RemoteSettings()
+
+    def test_get_recipes_is_empty(self):
+        rs = self.get_remote_settings()
+        assert rs.get_recipes() == []
+
+    def test_update_recipes(self):
+        rs = self.get_remote_settings()
+        new_recipes = [{"experiment1": True}, {"experiment2": False}]
+        rs.update_recipes(new_recipes)
+        assert rs.get_recipes() == new_recipes
+
+    @patch("requests.get")
```

Comment:
Oh I don't think this is the right pattern.  This is mocking out the `requests` object as its imported in this file, and then if this file is parsed by the Python runtime before it parses the Remote settings file, then that instance of the module in memory will be reused.  But for instance, if you moved 

```py
from ..cirrus.experiment_recipes import RemoteSettings
```

up above

```py
import requests
```

then `RemoteSettings` module would get initialized first, it would import its local copy of requests, then you'd import yours, then you'd be mocking the wrong one?

Generally you're supposed to mock the path to the object in the file that it's being called from, ie `cirrus.experiment_recipes.requests`

Reply:
makes sense 👍